java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -224,18 +225,31 @@ private static boolean emptyBody(MethodTree methodTree) {
   private static class MethodInvocationVisitor extends BaseTreeVisitor {
     private Set<Type> thrownExceptions = new HashSet<>();
     private boolean visitedUnknown = false;
+    private boolean visitedOtherConstructor = false;
+    private final MethodTree methodTree;
+
+    MethodInvocationVisitor(MethodTree methodTree) {
+      this.methodTree = methodTree;
+    }
 
     @Nullable
     public Set<Type> thrownExceptions() {
       if (visitedUnknown || thrownExceptions.stream().anyMatch(Type::isUnknown)) {
         // as soon as there is an unknown type, we discard any attempt to find an issue
         return null;
       }
+      if (methodTree.is(Tree.Kind.CONSTRUCTOR) && !visitedOtherConstructor) {
+        Optional<Symbol.MethodSymbol> constructor = getImplicitlyCalledConstructor(methodTree);
```

Comment:
I think it would be even cleaner if you do not use the variable:
```java
getImplicitlyCalledConstructor(methodTree)
  .map(Symbol.MethodSymbol::thrownTypes)
  .ifPresent(thrownExceptions::addAll);
```

Reply:
Yes, nice proposition!